From 90c9ae1a022373320b08427040470e1cb09ce1ba Mon Sep 17 00:00:00 2001
Message-Id: <90c9ae1a022373320b08427040470e1cb09ce1ba.1350297260.git.jdenemar@redhat.com>
From: "Daniel P. Berrange" <berrange@redhat.com>
Date: Wed, 10 Oct 2012 15:38:47 +0100
Subject: [PATCH] Remove pointless virLXCProcessMonitorDestroy method

https://bugzilla.redhat.com/show_bug.cgi?id=864336

Asynchronously setting priv->mon to NULL was pointless,
just remove the destroy callback entirely.

Signed-off-by: Daniel P. Berrange <berrange@redhat.com>
(cherry picked from commit dd0371764f90f31fa8e596b40c0269cdbd5082f6)
---
 src/lxc/lxc_process.c | 15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/src/lxc/lxc_process.c b/src/lxc/lxc_process.c
index 15572cd..b9cff85 100644
--- a/src/lxc/lxc_process.c
+++ b/src/lxc/lxc_process.c
@@ -554,20 +554,6 @@ cleanup:
 }
 
 
-static void virLXCProcessMonitorDestroy(virLXCMonitorPtr mon,
-                                        virDomainObjPtr vm)
-{
-    virLXCDomainObjPrivatePtr priv;
-
-    virDomainObjLock(vm);
-    priv = vm->privateData;
-    if (priv->monitor == mon)
-        priv->monitor = NULL;
-    if (virObjectUnref(vm))
-        virDomainObjUnlock(vm);
-}
-
-
 extern virLXCDriverPtr lxc_driver;
 static void virLXCProcessMonitorEOFNotify(virLXCMonitorPtr mon ATTRIBUTE_UNUSED,
                                           virDomainObjPtr vm)
@@ -649,7 +635,6 @@ static void virLXCProcessMonitorExitNotify(virLXCMonitorPtr mon ATTRIBUTE_UNUSED
 
 static virLXCMonitorCallbacks monitorCallbacks = {
     .eofNotify = virLXCProcessMonitorEOFNotify,
-    .destroy = virLXCProcessMonitorDestroy,
     .exitNotify = virLXCProcessMonitorExitNotify,
 };
 
-- 
1.7.12.3

